src/main/com/fulcrologic/rad/report.cljc

```diff
@@ -376,6 +376,27 @@
         (start-report! app report-class {:route-params route-params})
         (comp/transact! app [(dr/target-ready {:target report-ident})])))))
 
+(s/def ::arglist (s/and vector?
+                        (fn [[this props]]
+                          (and (not= this props)
+                               (not= this '_)
+                               (not= props '_)))))
+
+(s/def ::columns (s/coll-of symbol? :kind vector?))
+(s/def ::row-pk symbol?)
```

Comment:
This could be a symbol, but perhaps they know what things need to be in the attribute map for reports, and choose to use a map...that would still work in code, but you're not letting them do it.

Reply:
I hear you and agree. But again, I just copied your current 

```clojure
(req! &env sym options ::row-pk #(symbol? %))
```
which I believe also happens at macro-expansion time?!